horreum-integration-tests/src/test/java/io/hyperfoil/tools/horreum/it/HorreumClientIT.java

```diff
@@ -321,15 +321,18 @@ public void afterAll(QuarkusTestContext context) {
 
     private void instantiateClient() {
         if (horreumClient == null) {
-            String horreumBaseUrl = "http://localhost:".concat(System.getProperty("quarkus.http.test-port"));
+            String testPort = System.getProperty("quarkus.http.test-port");
+            if (horreumBaseUrl != null ) {
+               horreumBaseUrl = "http://localhost:".concat(testPort);
+            } else {     
             horreumClient = new HorreumClient.Builder()
```

Comment:
This `if`/`else` block has changed the logic of the code. The `horreumClient = new HorreumClient.Builder()` section should nto be in the `else` block, otherwise we might not initialize the client

Reply:
you mean?
```java
private void instantiateClient() {
        if (horreumClient == null) {
            String horreumBaseUrl = "http://localhost:".concat(System.getProperty("quarkus.http.test-port"));
        } else {
            horreumClient = new HorreumClient.Builder()
                    .horreumUrl(horreumBaseUrl + "/")
                    .horreumUser("user")
                    .horreumPassword("secret")
                    .build();

            Assertions.assertNotNull(horreumClient);

      }  
    }
```

Reply:
What are you trying to do here?

Reply:
> The `horreumClient = new HorreumClient.Builder()` section should nto be in the `else` block, otherwise we might not initialize the client

Reply:
I want to fix this: https://github.com/Hyperfoil/Horreum/actions/runs/7236371643/job/19714850280#step:6:2524

Reply:
This is the root problem: https://github.com/Hyperfoil/Horreum/actions/runs/7236371643/job/19714850280#step:6:2473